packages/twenty-server/src/engine/core-modules/billing/services/billing-subscription.service.ts

```diff
@@ -164,42 +165,75 @@ export class BillingSubscriptionService {
         ? SubscriptionInterval.Month
         : SubscriptionInterval.Year;
 
-    const billingSubscriptionItem =
-      await this.getCurrentBillingSubscriptionItemOrThrow(workspace.id);
-
-    let productPrice;
+    const billingBaseProductSubscriptionItem =
+      await this.getCurrentBaseProductBillingSubscriptionItemOrThrow(
+        workspace.id,
+      );
 
     if (isBillingPlansEnabled) {
-      const baseProduct = await this.billingPlanService.getPlanBaseProduct(
-        BillingPlanKey.PRO,
+      const pricesPerPlan = await this.billingPlanService.getPricesPerPlan({
+        planKey: BillingPlanKey.PRO,
+        interval: newInterval,
+      });
+      const billingPricesPerPlanArray = [
+        pricesPerPlan.baseProductPrice,
+        ...pricesPerPlan.meteredProductsPrices,
+      ].filter((price): price is BillingPrice => price !== undefined);
+
+      const updatedSubscriptionItems = this.getUpdatedSubscriptionItems(
+        billingSubscription,
+        billingPricesPerPlanArray,
       );
 
-      if (!baseProduct) {
-        throw new BillingException(
-          'Base product not found',
-          BillingExceptionCode.BILLING_PRODUCT_NOT_FOUND,
-        );
-      }
-
-      productPrice = baseProduct.billingPrices.find(
-        (price) => price.interval === newInterval,
+      await this.stripeSubscriptionService.updateSubscriptionItems(
+        billingSubscription.stripeSubscriptionId,
+        updatedSubscriptionItems,
       );
     } else {
-      productPrice = await this.stripePriceService.getStripePrice(
+      const productPrice = await this.stripePriceService.getStripePrice(
         AvailableProduct.BasePlan,
         newInterval,
       );
-    }
 
-    if (!productPrice) {
-      throw new Error(
-        `Cannot find product price for product ${AvailableProduct.BasePlan} and interval ${newInterval}`,
+      if (!productPrice) {
+        throw new Error(
+          `Cannot find product price for product ${AvailableProduct.BasePlan} and interval ${newInterval}`,
+        );
+      }
+
+      await this.stripeSubscriptionItemService.updateBillingSubscriptionItem(
+        billingBaseProductSubscriptionItem,
+        productPrice.stripePriceId,
       );
     }
+  }
+
+  private getUpdatedSubscriptionItems(
+    billingSubscription: BillingSubscription,
+    billingPricesPerPlanArray: BillingPrice[],
+  ): Stripe.SubscriptionItemUpdateParams[] {
+    return billingSubscription.billingSubscriptionItems.map(
```

Comment:
It could help to have a const updateParams = ....map(...)

Then return updateParams.